fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -143,16 +128,54 @@ where
         atomically(|| self.provider.cached_blocks()).await > max_cache_blocks
     }
 
-    /// Poll the next block height. Adds the finalized and executed block data to the caches.
-    ///
-    /// Returns the height which was polled.
-    ///
-    /// This method is only expected to be called after the caller has checked that the chain
-    /// is supposed to have data at the next height.
-    async fn poll_next(&mut self) -> Result<BlockHeight, Error> {
-        let height = self.sync_pointers.head() + 1;
-        let parent_block_hash = self.non_null_parent_hash().await;
+    /// Get the data needed to pull the next block
+    async fn get_from_cache(&self) -> (BlockHeight, BlockHash) {
+        // we are getting the latest height fetched in cache along with the first non null block
+        // that is stored in cache.
+        // we are doing two fetches in one `atomically` as if we get the data in two `atomically`,
+        // the cache might be updated in between the two calls. `atomically` should guarantee atomicity.
```

Comment:
Just wondering: who else might be updating the cache?

Reply:
when the finality is committed, some of the cache might be purged